x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference;
+
+import org.elasticsearch.features.FeatureSpecification;
+import org.elasticsearch.features.NodeFeature;
+import org.elasticsearch.xpack.inference.rank.textsimilarity.TextSimilarityRankRetrieverBuilder;
+
+import java.util.Set;
+
+/**
+ * Provides inference features.
+ */
+public class InferenceFeatures implements FeatureSpecification {
+
+    @Override
+    public Set<NodeFeature> getFeatures() {
+        return Set.of(TextSimilarityRankRetrieverBuilder.TEXT_SIMILARITY_RERANKER_RETRIEVER_SUPPORTED);
```

Comment:
@benwtrent @pmpailis I added this feature spec to the `inference` plugin so that we can [check the feature and fail fast in a mixed-cluster scenario](https://github.com/elastic/elasticsearch/pull/109813/files#diff-d1390fdfd7cd709bf892434e8fe8192cd055e8b55efd77174fa580d88e2542a8R66-R68).